xarray/core/indexing.py

```diff
@@ -1085,10 +1091,17 @@ def _masked_result_drop_slice(key, chunks_hint=None):
             else k
             for k in key
         ]
-    return _logical_any(k == -1 for k in key)
+
+    mask = _logical_any(k == -1 for k in key)
+    if isinstance(data, sparse_array_type):
+        import sparse
+
+        mask = sparse.COO.from_numpy(mask)
```

Comment:
I think it's probably better to convert the indexer before logical_any above. If I understand correctly, the indexer keys will generally be 1d but the mask may be higher dimensional, like the data itself.

Reply:
OK the latest commit does that.